working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1718,20 +1706,19 @@ To type check a pattern `p` being matched against a value of type `M`:
     1.  Resolve the extractor name to a type `X`. It is a compile-time error if
         the name does not refer to a type. Apply downwards inference from `M`
         to infer type arguments for `X` if needed.
-
-    1.  Type-check each of `f`'s field subpatterns using the type of the getter
-        on `X` with the same name as the field as the matched value type. It is
-        a compile-time error if `X` does not have a getter whose name matches
-        the subpattern's field name.
-
-    2.  The required type of `p` is `X`.
+2.  For each field subpattern of `p`, with name `n` and subpattern `f`:
+        1.  Let `G` be the the type of the getter on `X` with the name `n`. 
+            It is a **compile-time error** if `X` does not have a *getter* with name `n`.
+            _If `X` is `dynamic` or `Never`, it is considered as having every getter with the same type._
+        2.  Type check `f` with `G` as the matched value type, to find its required type.
+3.  The required type of `p` is `X`.
 
 It is a compile-time error if:
 
 *   The type of an expression in a guard clause is not assignable to `bool`.
-
-*   `p` is in an irrefutable context, it has a required type `T`, and `M` is not
-    assignable to `T`. *Destructuring and variable patterns can only be used in
+*   A pattern `p` is in an irrefutable context (directly or nested as a subpattern), 
```

Comment:
The proposal does specify that subpatterns are already in the same context as the outermost pattern so this parenthetical is redundant, but not harmful:

> *   **Declaration context.** The pattern in `localVariableDeclaration`,
    `forLoopParts`, or any of its subpatterns. ...
> *   **Assignment context.** The pattern in a `patternAssignment` or any of its
    subpatterns. ...
> *   **Matching context.** The pattern in a `caseHead` or any of its subpatterns. ...

Reply:
Might as well remove this to be consistent.

(I'm trying to find a way to nest a pattern inside another pattern, without it being a subpattern, but we don't allow general expressions inside a pattern, only constants which cannot contain patterns, so it probably can't happen.)